engine/sound/src/sound.cpp

```diff
@@ -891,6 +894,16 @@ namespace dmSound
         SoundGroup* group = &sound->m_Groups[*index];
         group->m_Gain.Set(GainToScale(gain), reset);
         group->m_GainParameter = gain;
+        if (gain > 0.0f)
+        {
+            group->m_GainBeforeMute = gain;
+            group->m_IsMuted = false;
+        }
+        else
+        {
+            group->m_IsMuted = true;
+        }
```

Comment:
Overall, I feel that the if-statements surrounding this logic (>0.0f) is a bit redundant, and hinders readability. `gain = 0` is a valid value.

E.g. what about:
```c++
group->m_GainBeforeMute = gain;
group->m_IsMuted = group->m_GainBeforeMute > 0.0f;
```